api/src/users/zod/user.ts

```diff
@@ -0,0 +1,21 @@
+import { UserRole } from '@prisma/client';
+import { z } from 'zod'
+
+export const $CreateUserDto = z.object({
+
+    confirmedAt: z.coerce.date().optional(),
+
+    email: z.string().email(),
+
+    firstName: z.string().min(1),
+
+    lastName: z.string().min(1),
+
+    password: z.string(),  // TO_DO: let user configure how strong the password should be
```

Comment:
Don't worry about this for now. We can implement this in JSLibs since I need for OpenDataCapture as well